.github/workflows/release.yml

```diff
@@ -0,0 +1,68 @@
+name: Release
+
+on:
+  pull_request:
+    types: [closed]
+    paths:
+      - '.github/project.yml'
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    name: release
+    if: ${{github.event.pull_request.merged == true}}
+
+    steps:
+      - uses: radcortez/project-metadata-action@master
+        name: Retrieve project metadata
+        id: metadata
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          metadata-file-path: '.github/project.yml'
+
+      - uses: actions/checkout@v2
+
+      - name: Import GPG key
+        id: import_gpg
+        uses: crazy-max/ghaction-import-gpg@v3
+        with:
+          gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.GPG_PASSPHRASE }}
+
+      - name: Install JDK 11
+        uses: joschi/setup-jdk@e87a7cec853d2dd7066adf837fe12bf0f3d45e52
+        with:
+          distribution: 'temurin'
+          java-version: 11
+          check-latest: true
+
+      - name: Configure Git author
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+      - name: Maven release ${{steps.metadata.outputs.current-version}}
+        run: |
+          mvn clean install -DskipTests
+          gpg --quiet --batch --yes --decrypt --passphrase="${{secrets.GPG_PASSPHRASE}}" --output maven-settings.xml .github/release/maven-settings.xml.gpg
+          git checkout -b release
+          mvn -X -Prelease -B release:clean release:prepare -DreleaseVersion=${{steps.metadata.outputs.current-version}} -DdevelopmentVersion=${{steps.metadata.outputs.next-version}} -s maven-settings.xml
```

Comment:
Using the maven release plugin, means that we somehow need to log in to sonatype to close/release the repo, right?
This is no biggie, but this still goes through my account. So, we need to see what needs to be done so that you can have access to the repo on oss sonatype.

Reply:
The maven release will manage the connection with sonatype. And this is why gpg is needed: we need to write one Maven settings file with your OSS account (or the token) and encrypt it with a gpg key (that will be stored in repo secrets). 
I think we can get rid of the gpg setup and store the sonatype credentials directly to secrets though.

Reply:
What I am trying to say, is that this procedure still requires manual steps to be performed post release. It would be nice if we could avoid it (but for now its not a blocker).